src/Blueprint/Blueprint/Cmdlets/BlueprintCmdletBase.cs

```diff
@@ -0,0 +1,382 @@
+        {
+            var localAssignment = new Assignment
+            {
+                Identity = new ManagedServiceIdentity { Type = identityType },
+                Location = bpLocation,
+                BlueprintId = blueprintId,
+                Locks = new AssignmentLockSettings { Mode = lockMode == null ? PSLockMode.None.ToString() : lockMode.ToString() },
+                Parameters = new Dictionary<string, ParameterValueBase>(),
+                ResourceGroups = new Dictionary<string, ResourceGroupValue>()
+            };
+
+            if (Parameters != null)
+            {
+                foreach (var key in Parameters.Keys)
+                {
+                    var value = new ParameterValue(Parameters[key], null);   
+                    localAssignment.Parameters.Add(key.ToString(), value);
+                }   
+            }
+
+            if (ResourceGroups != null)
+            {
+                foreach (var key in ResourceGroups.Keys)
+                {
+                    var kvp = ResourceGroups[key] as Hashtable;
+                    string name = null;
+                    string location = null;
+
+                    foreach (var k in kvp.Keys)
+                    {
+                        var rgKey = k.ToString();
+
+                        if (string.Equals(rgKey, "name", StringComparison.InvariantCultureIgnoreCase))
+                        {
+                            name = kvp[k].ToString();
+                        }
+
+                        if (string.Equals(rgKey, "location", StringComparison.InvariantCultureIgnoreCase))
+                        {
+                            location = kvp[k].ToString();
+                        }
+                    }
+
+                    var rgv = new ResourceGroupValue(name, location);
+                    localAssignment.ResourceGroups.Add(key.ToString(), rgv);
+                }
+            }
+
+            return localAssignment;
+        }
+
+        /// <summary>
+        /// Register Blueprint RP with a subscription.
+        /// </summary>
+        /// <param name="subscriptionId"> SubscriptionId passed from the cmdlet</param>
+        protected void RegisterBlueprintRp(string subscriptionId)
+        {
+            ResourceManagerClient.SubscriptionId = subscriptionId;
+            ResourceManagerClient.Providers.Register(BlueprintConstants.BlueprintProviderNamespace);
+        }
+
+        /// <summary>
+        /// Get Blueprint SPN for this tenant
+        /// </summary>
+        /// <returns></returns>
+        protected ServicePrincipal GetBlueprintSpn()
+        {
+            var odataQuery = new Rest.Azure.OData.ODataQuery<ServicePrincipal>(s => s.ServicePrincipalNames.Contains(BlueprintConstants.AzureBlueprintAppId));
+            var servicePrincipal = GraphRbacManagementClient.ServicePrincipals.List(odataQuery.ToString())
+                .FirstOrDefault();
+
+            return servicePrincipal;
+        }
```

Comment:
This only really applies to cmdlets that create or upgrade an assignment. Should we move that over to some base class somewhere?

Reply:
Also the method below

---
In reply to: [257093525](https://github.com/Azure/azure-powershell/pull/8537#discussion_r257093525) [](ancestors = 257093525)

Reply:
Just to check, this is in the cmdletbase class. Do you think it'd be better if we had a base class for create/upgrade operations?

---
In reply to: [257093525](https://github.com/Azure/azure-powershell/pull/8537#discussion_r257093525) [](ancestors = 257093525)

Reply:
I was thinking about moving that into an Assignment specific base class somewhere. Any assignment related cmdlet would need this logic. (Unless I'm wrong and there are other usages.)

---
In reply to: [257326054](https://github.com/Azure/azure-powershell/pull/8537#discussion_r257326054) [](ancestors = 257326054,257093525)

Reply:
Sounds good. I'll work on that in the next phase of the cmdlets.

---
In reply to: [257382430](https://github.com/Azure/azure-powershell/pull/8537#discussion_r257382430) [](ancestors = 257382430,257326054,257093525)